emission/storage/decorations/timeline.py

```diff
@@ -49,6 +49,34 @@ def get_timeline(user_id, start_ts, end_ts):
 
     return Timeline(places, trips)
 
+def get_aggregate_timeline_from_dt(start_dt, end_dt, box=None):
+    import emission.core.get_database as edb
+    import emission.core.wrapper.entry as ecwe
+    import emission.storage.decorations.place_queries as esdp
+    import emission.storage.decorations.trip_queries as esdt
+
+    if not box:
+        logging.info("About to query for %s -> %s" % (start_dt, end_dt))
+    else:
+        logging.info("About to query for %s -> %s in %s" % (start_dt, end_dt, box))
+    result_cursor = edb.get_timeseries_db().find({"data.local_dt": {"$gte": start_dt, "$lte": end_dt}}).sort("metadata.write_ts")
+    logging.debug("result cursor has %d entries" % result_cursor.count())
+    result_list = list(result_cursor)
+    logging.debug("result list has %d entries" % len(result_list))
+    if len(result_list) == 0:
+        return Timeline([], [])
+    start_ts = ecwe.Entry(result_list[0]).metadata.write_ts
+    end_ts = ecwe.Entry(result_list[-1]).metadata.write_ts
+    logging.debug("Converted datetime range %s -> %s to timestamp range %s -> %s" %
+        (start_dt, end_dt, start_ts, end_ts))
+    if box:
+       places = esdp.get_aggregate_places(enua.UserCache.TimeQuery("enter_ts", start_ts, end_ts), box=box)
+       trips = esdt.get_aggregate_trips(enua.UserCache.TimeQuery("start_ts", start_ts, end_ts), box=box)
```

Comment:
I don't think that you need this check. You can just pass box=box through and if the input was None, this will be None as well.